### PR TITLE
[Feature][Flink] Support parallelism input

### PR DESCRIPTION
**Pull request by chengshiwen**
## What is the purpose of the pull request

*Support parallelism input in Flink*

This pr is just an improvement of the `parallelism` of the issue #4924

## Brief change log

  - *Support parallelism input in Flink in UI*
  - *Support parallelism input in Flink in Server*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

  - *dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/utils/FlinkArgsUtilsTest.java*

(and)

This change added tests and can be verified as follows:

  - *Manually verified the change by testing locally.*

In cluster mode:
image

In local mode:
image

### CODE REVIEW

**Review by break60 — APPROVED** [review]
front end +1
good job

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java [hunk 1]
```diff
@@ -852,8 +854,9 @@ private Constants() {
 
     public static final String FLINK_JOB_MANAGE_MEM = "-yjm";
     public static final String FLINK_TASK_MANAGE_MEM = "-ytm";
-    public static final String FLINK_DETACH = "-d";
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:855]: hi, I don't know why need to delete this param, I read the issue of this PR, he said above should also need to support this parameter.

Comment (chengshiwen) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:855]: -  `-d,--detached`, `-s,--fromSavepoint`, `--allowNonRestoredState` are optional and not common options, so they are more suitable to be placed in the `option parameters`.
- `-p,--parallelism` is a common and frequently used option, so I add it to the code.
- `-d,--detached` is not a good option for DolphinScheduler, which means `runs the job in detached mode`. This mode will cause that the task state and the yarn job state are different. We will see that the DolphinScheduler task ends early, but in fact the task on yarn is still running.
- ` -sae,--shutdownOnAttachedExit` is better, which means `If the job is submitted in attached mode, perform a best-effort cluster shutdown when the CLI is terminated abruptly`. So I just replace `-d` with `-sae`, and this feature has been verified in our company for 3 months.
- The proposals of the author of that pr are not all good, just both `Generic CLI` mode and  `-p,--parallelism` are good. This feature `-p, --parallelism` is what I wanted to add two months ago

Comment (CalvinKirs) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:855]: Thx,I got it

**Review by CalvinKirs — APPROVED** [review]
cool
LGTM

**Pull request merged**